core/src/main/java/com/megaease/easeagent/core/Register.java

```diff
@@ -36,15 +36,14 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(Register.class);
 
     private final Iterable<QualifiedBean> beans;
-    private final Set<Integer> applied;
+    private final Set<String> applied = new HashSet<>();
 
     Register(Iterable<QualifiedBean> beans) {
         this.beans = beans;
-        applied = new HashSet<Integer>();
     }
 
     void apply(String adviceClassName, ClassLoader external) {
-        if (!applied.add(Objects.hashCode(external, adviceClassName))) return;
+        if (!applied.add(adviceClassName + external.getClass().getName())) return;
```

Comment:
This method has a problem:
```
adviceClassName = "111" , ClassLoader external = "112"  key="111112"
adviceClassName = "1111" , ClassLoader external = "12"  key="111112"

```
We should use a separator like `-`.


```
separator="-"
adviceClassName = "111" , ClassLoader external = "112"  key="111-112"
adviceClassName = "1111" , ClassLoader external = "12"  key="1111-12"

```

Reply:
updated. and just found that it's incorrect to use the class name of the `external`.